### PR TITLE
Render docstrings and datatable cells with example table entries

### LINKED ISSUE #252: Scenario Outline that contains placeholder in its name
Is there a way to provide a parameter to Scenario Outline step definition so we can put detailed description to each scenario?
I am asking for something similar to this:
https://behave.readthedocs.io/en/latest/new_and_noteworthy_v1.2.5.html#name-may-contain-placeholders
```
Scenario Outline: Eating: <case>
    Given there are <start> cucumbers
    When I eat <eat> cucumbers
    Then I should have <left> cucumbers

    Examples:
    | case    | start | eat | left |
    | Eat few | 12    | 5   | 7    |
    | Eat all | 12    | 12  | 0    |
```

Above gives error:
```
pytest_bdd.exceptions.ScenarioExamplesNotValidError: Scenario "Eating: <case>" in the feature "/pytest_bdd_issue/outline.feature" has not valid examples. Set of step parameters ['eat', 'left', 'start'] should match set of example values ['case', 'eat', 'left', 'start'].
```
If I add `case` to one of the steps then it runs but in the report it doesn't fill the placeholder in Scenario Outline.

I have created repo with failing example:
https://github.com/sliwinski-milosz/pytest-bdd-issue

### LINKED ISSUE #741: Cannot parse datatable with examples
Sample Scenario

```
Scenario Outline: User selects bookmark position
    When the user selects A in the bookmarks
    Then the position should be the following
      | x   | y   | z   |
      | <x> | <y> | <z> |
```

    Examples:
      | x | y | z |
      | 1 | 2 | 3 |

Before, it could parse the values correctly, returning 1, 2, 3, but now it returns the placeholders as strings `<x>, <y>, <z>.`
This worked before the recent change in data tables.
Hope you could help check this issue, thank you

### PR DESCRIPTION
**Pull request by jsa34**
Render scenario names, docstrings and datatable cells with example table entries, just like step names currently are.

Fixed another issue that I hadn't seen before that background steps were also not able to be rendered using the current logic.  This is now fixed.

Resolves #741 
Resolves #252

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -513,6 +513,59 @@ Example:
...
+            Given the following configuration:
+                """
+                username: <username>
+                password: <password>
+                """
+            When the user logs in
+            Then the response should contain:
+                | field     | value       |
+                | username  | <username> |
+                | logged_in | true        |
```
Comment (vianney-g) [inline README.rst:538]:
```suggestion
                | field     | value      |
                | username  | <username> |
                | logged_in | true       |
```

Comment (jsa34) [reply README.rst]: Thanks, @vianney-g !

**Comment by youtux** [comment]
hey I took the liberty of applying some changes directly, hope you don't mind. If that's fine to you, I'd merge it.

**Review by youtux — APPROVED** [review]

**Comment by jsa34** [comment]
> hey I took the liberty of applying some changes directly, hope you don't mind. If that's fine to you, I'd merge it.

Not at all!  I did notice when I code reviewed that the STEP_PARAM_RE regex is no longer just for steps so renamed it.

**Comment by jsa34** [comment]
I can't merge :(. Sorry my new commit invalidated your approval @youtux

**Comment by youtux** [comment]
turns out that the `Step.params` method is unused, so I removed it.

**Review by youtux — APPROVED** [review]

**Pull request merged**